### PR TITLE
feat: [ALIGN-988] Add feedback collection feature to Kotlin SDK

### PR DESCRIPTION
**Pull request by bosungpark**
## Background

As with other programming languages(ex. python, node), we are updating the Kotlin SDK to include feedback-related features.

## Changes

1. Added feedback collection functionality to the Kotlin SDK.
2. Fixed build failure issues.
3. Updated buffer schema dependencies.

스크린샷 2024-11-19 오후 12 22 19

### CODE REVIEW

File: src/main/kotlin/com/github/coxwave/alignsdkkotlin/AlignAI.kt [hunk 1]
```diff
@@ -39,10 +40,11 @@ class AlignAI(
                 "Authorization" to listOf("Bearer ${config.apiKey}")
             ),
         )
-        if (response.code == Code.OK) {
+
```
Comment (bosungpark) [inline src/main/kotlin/com/github/coxwave/alignsdkkotlin/AlignAI.kt:42]: Fixed build failure issues.

File: build.gradle.kts [hunk 2]
```diff
@@ -19,11 +19,11 @@ dependencies {
...
     implementation("org.jetbrains.kotlinx:kotlinx-coroutines-core:1.7.3")
-    implementation("com.connectrpc:connect-kotlin-okhttp:0.5.0")
-    implementation("com.connectrpc:connect-kotlin-google-java-ext:0.5.0")
+    implementation("com.connectrpc:connect-kotlin-okhttp:0.6.0")
+    implementation("com.connectrpc:connect-kotlin-google-java-ext:0.6.0")
     implementation("com.google.protobuf:protobuf-kotlin:3.25.3")
-    implementation("build.buf.gen:impaction-ai_ingestion_connectrpc_kotlin:0.5.0.1.20240304081353.ab10776f8380")
-    implementation("build.buf.gen:impaction-ai_ingestion_protocolbuffers_kotlin:25.3.0.1.20240304081353.ab10776f8380")
+    implementation("build.buf.gen:impaction-ai_ingestion_connectrpc_kotlin:0.7.1.1.20240908234409.342910308823")
+    implementation("build.buf.gen:impaction-ai_ingestion_protocolbuffers_kotlin:28.3.0.2.20240908234409.342910308823")
```
Comment (kde713) [inline build.gradle.kts:26]: https://buf.build/protocolbuffers/kotlin?version=v28.3

protocolbuffers/kotlin plugin v28.3 recommends to set dependency version to v4.28.3 for protobuf-kotlin. Can you double-check the dependency compatibility?

File: build.gradle.kts [hunk 3]
```diff
@@ -19,11 +19,11 @@ dependencies {
...
     implementation("com.squareup.okhttp3:okhttp:4.12.0")
     implementation("org.jetbrains.kotlinx:kotlinx-coroutines-core:1.7.3")
-    implementation("com.connectrpc:connect-kotlin-okhttp:0.5.0")
-    implementation("com.connectrpc:connect-kotlin-google-java-ext:0.5.0")
+    implementation("com.connectrpc:connect-kotlin-okhttp:0.6.0")
+    implementation("com.connectrpc:connect-kotlin-google-java-ext:0.6.0")
     implementation("com.google.protobuf:protobuf-kotlin:3.25.3")
-    implementation("build.buf.gen:impaction-ai_ingestion_connectrpc_kotlin:0.5.0.1.20240304081353.ab10776f8380")
-    implementation("build.buf.gen:impaction-ai_ingestion_protocolbuffers_kotlin:25.3.0.1.20240304081353.ab10776f8380")
+    implementation("build.buf.gen:impaction-ai_ingestion_connectrpc_kotlin:0.7.1.1.20240908234409.342910308823")
```
Comment (kde713) [inline build.gradle.kts:25]: https://buf.build/connectrpc/kotlin?version=v0.7.1

connectrpc/kotlin plugin v0.7.1 recommends to set dependency version to `0.7.1` for connect-kotlin, connect-kotlin-okhttp, connect-kotlin-google-java-ext. Can you double-check the dependency compatibility?

File: src/main/kotlin/com/github/coxwave/alignsdkkotlin/AlignAI.kt [hunk 4]
```diff
@@ -3,7 +3,8 @@ package com.github.coxwave.alignsdkkotlin
 import com.github.coxwave.alignsdkkotlin.events.BaseEvent
 import build.buf.gen.ingestion.v1alpha.CollectEventsRequest
 import build.buf.gen.ingestion.v1alpha.IngestionServiceClient
-import com.connectrpc.Code
+import com.connectrpc.ResponseMessage
+
```
Comment (kde713) [inline src/main/kotlin/com/github/coxwave/alignsdkkotlin/AlignAI.kt]: Unnecessary blank line should be removed.

File: build.gradle.kts (hunk 3 above)
Comment (bosungpark) [reply build.gradle.kts:25]: Confirmed. The dependency versions you mentioned are correct. Thanks a lot!

File: build.gradle.kts (hunk 2 above)
Comment (bosungpark) [reply build.gradle.kts:26]: Confirmed. The dependency versions you mentioned are correct. Thanks a lot!

**Review by kde713 — COMMENTED** [review]
README.md includes a guide for dependencies that needs to be updated.

**Comment by bosungpark** [comment]
> README.md includes a guide for dependencies that needs to be updated.

Thank you, Updated README.md

File: README.md [hunk 5]
```diff
@@ -31,15 +31,15 @@ dependencies {
     ...
 
     // SDK
-    implementation("com.github.coxwave:alignai-sdk-kotlin:0.3.0")
```
Comment (bosungpark) [inline README.md:34]: Update sdk version

**Review by kde713 — APPROVED** [review]
LGTM!

**Pull request merged**